### PR TITLE
Add script for accuracy validation

### LINKED ISSUE #5: Validation for converted models for accuracy
The goal would be to get identical inference results to Paddle models and inference. @kuke has more thoughts. And here is a notebook on the topic: https://github.com/onnx/tutorials/blob/master/tutorials/CorrectnessVerificationAndPerformanceComparison.ipynb.

Q: what runtime would we use for ONNX: TensorRT?

### PR DESCRIPTION
**Pull request by kuke**
Resolve #5 

To run this validation script, need to install Caffe2 to use its ONNX backend for the backend in the [onnx/onnx](https://github.com/onnx/onnx) repo has a bug when executing ```run_model()```. And at the same time, need to install the latest ONNX from source, otherwise some other problems will happen.

### CODE REVIEW

File: validate.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    [fluid_infer_program, feed_target_names,
+     fetch_targets] = fluid.io.load_inference_model(args.fluid_model, exe)
+
+    input_shapes = [
+        fluid_infer_program.global_block().var(var_name).shape
+        for var_name in feed_target_names
+    ]
+    input_shapes = [
+        shape if shape[0] > 0 else (args.batch_size, ) + shape[1:]
```
Comment (varunarora) [inline validate.py]: Does this use `paddle_onnx_shape` or is this completely different

**Comment by pkuyym** [comment]
Excellent Job. Some questions:
1. FC --> matmul ? why not FC --> FC ?
2. Current weight `FC.W` and `FC.b` using a 'Constant' type, I think we should consider this field 
`https://github.com/onnx/onnx/blob/master/onnx/onnx-ml.proto3#L242` which should be a general solution.

File: validate.py (hunk 1 above)
Comment (kuke) [reply validate.py]: ```paddle_onnx_shape``` only changes the unknown dim for -1 to 0 (required by ONNX). And this line is used to set the batch size.

**Comment by kuke** [comment]
1. No `FC` op in Fluid

2. I cannot find the proper way to initialize tensor, and using `Constant` op to store parameters makes sense.

File: validate.py [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import argparse
+import numpy as np
+
+import paddle.fluid as fluid
+from onnx import helper, checker, load
+from caffe2.python.onnx.backend import Caffe2Backend
```
[the remark this answers is not in the document]
Comment (kuke) [reply validate.py]: Caffe2 cannot be installed by ```pip``` and [its installation](https://caffe2.ai/docs/getting-started.html?platform=ubuntu&configuration=prebuilt) is not that direct without Anaconda environment.

File: validate.py [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+
+def parse_args():
+    # Read arguments: path to model.
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--fluid_model",
+        required=True,
+        help="The path to PaddlePaddle Fluid model.")
+    parser.add_argument(
```
[the remark this answers is not in the document]
Comment (kuke) [reply validate.py]: OK. I will modify the argument in #9

File: validate.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (varunarora) [reply validate.py]: Got it, just add some notes to the README then

[the remark this answers is not in the document]
Comment (kuke) [reply validate.py]: OK

**Comment by varunarora** [comment]
Hey @kuke, isnt there an FC in fluid? http://paddlepaddle.org/docs/develop/api/fluid/en/layers.html#permalink-36-fc

**Comment by kuke** [comment]
@varunarora No. The `fc` layer consists of `mul` op and bias `add` op in Fluid

https://github.com/PaddlePaddle/Paddle/blob/a64edbf14e9c2f499a0dbb34417c50b8ded67c58/python/paddle/fluid/layers/nn.py#L192

**Pull request merged**